js/packages/cli/src/commands/upload.ts

```diff
@@ -102,7 +105,7 @@ export async function upload(
             symbol: manifest.symbol,
             sellerFeeBasisPoints: manifest.seller_fee_basis_points,
             isMutable: mutable,
-            maxSupply: new BN(0),
+            maxSupply: new BN(max_supplies[i.toString()]),
```

Comment:
you realize this is only run once, right? Look at the if statement. i===0. So you can only set maxSupply once for the entire config - all share the same one. So there is no point in loading a file here - just take it as a single pass in option.